### ISSUE TITLE
[quotes] Clarify that "avoid-escape" does not allow backticks

### ISSUE THREAD

**Issue by feross** [eslint/eslint#5147]
Using the latest eslint (2.0.0-rc.0).

Rules:

```json
{
  "parserOptions": {
    "ecmaVersion": 6
  },
  "rules": {
    "quotes": [2, "single", "avoid-escape"]
  }
}
```

Code:

```js
console.log(`hello 'hi'`)
```

Gives the following false-positive error:

```
  1:13  error  Strings must use singlequote  quotes
```

The ` string should be allowed, because it's being used to avoid quotes.

Changing the code to:

```js
console.log("hello 'hi'")
```

causes the error to go away. This is inconsistent and contrary to what the docs say.

**Comment by eslintbot** [comment]
@feross Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by feross** [comment]
It's also worth noting that this worked correctly in 1.x.

**Comment by IanVS** [comment]
I believe the reason for this is that backticks are only allowed when they are not the chosen quote type if they add value.  See https://github.com/eslint/eslint/issues/3090.  If you want to avoid escaping the single quotes, you should use double quotes.  

The docs contain the line:

> When using `"single"` or `"double"`, template literals that don't contain a substitution, don't contain a line break, and aren't tagged templates are flagged as problems.

But perhaps it could probably be explained more clearly with regards to "avoid-escape".

**Comment by feross** [comment]
Ah, I see. Thanks for explaining. I agree with the rationale.

This was a breaking change for several packages that use `standard` because they were using backticks to avoid escaping. But we're going to release a new major version (6.0.0), so I think this is okay.

**Comment by IanVS** [comment]
Let's keep this open so we remember to clarify in the docs that `"avoid-escape"` only applies to single quotes and double quotes.

**Comment by charlesdeuter** [comment]
> If you want to avoid escaping the single quotes, you should use double quotes.

Many of code bases I work with prefer backticks over double quotes for avoiding single quote escaping. It makes no sense to me that eslint would be opinionated in this reqard. This should be user configurable. I view this as a bug.
